### ISSUE TITLE
generating a new password takes a long time

### ISSUE THREAD

**Issue by michaldejmek** [nilsteampassnet/TeamPass#3606]
version 3.0.0.22 - I need to create a new password for a user, but the change takes a long time.

Encrypting keys - Logs [5000 - 5100] ... Please wait! This is where it ends :(

Isn't it possible to have the password generation run in the background to send the password to the user in an email once everything is done . ?

This problem was already reported last year :(

**Comment by AKorolkovs** [comment]
agree. Changing password takes up to 30 minutes. It is way too much for such simple thing

**Comment by wease1** [comment]
watch here
https://github.com/nilsteampassnet/TeamPass/issues/3440

**Comment by ftoledo** [comment]
Perhaps you can implement a configuration option that you choose which is the method to generate the password.
The current method or pipe to a command. I usually use pwgen (https://github.com/tytso/pwgen) that comes via apt-get in debian for example. I really know if it is safe like the method that tempass uses, but the creation is instantaneous and can be used to many cases.

**Comment by nilsteampassnet** [comment]
Closed automatically due to inactivity (no response for over 1 year).
